tidy3d/components/geometry.py

```diff
@@ -3280,8 +3289,374 @@ def _surface_area(self, bounds: Bound) -> float:
+        # currently ignores bounds
+        return self.trimesh.area
+
+    def _volume(self, bounds: Bound) -> float:
+        """Returns object's volume given bounds."""
+        # currently ignores bounds
+        return self.trimesh.volume
+
+    @cached_property
+    def bounds(self) -> Bound:
+        """Returns bounding box min and max coordinates.
+
+        Returns
+        -------
+        Tuple[float, float, float], Tuple[float, float float]
+            Min and max bounds packaged as ``(minx, miny, minz), (maxx, maxy, maxz)``.
+        """
+        if self.mesh_dataset is None:
+            return ((-inf, -inf, -inf), (inf, inf, inf))
+        return self.trimesh.bounds
+
+    def intersections_plane(
+        self, x: float = None, y: float = None, z: float = None
+    ) -> List[Shapely]:
+        """Returns list of shapely geoemtries at plane specified by one non-None value of x,y,z.
+
+        Parameters
+        ----------
+        x : float = None
+            Position of plane in x direction, only one of x,y,z can be specified to define plane.
+        y : float = None
+            Position of plane in y direction, only one of x,y,z can be specified to define plane.
+        z : float = None
+            Position of plane in z direction, only one of x,y,z can be specified to define plane.
+
+        Returns
+        -------
+        List[shapely.geometry.base.BaseGeometry]
+            List of 2D shapes that intersect plane.
+            For more details refer to
+            `Shapely's Documentaton <https://shapely.readthedocs.io/en/stable/project.html>`_.
+        """
+
+        if self.mesh_dataset is None:
+            return []
+
+        axis, position = self.parse_xyz_kwargs(x=x, y=y, z=z)
+
+        origin = self.unpop_axis(position, (0, 0), axis=axis)
+        normal = self.unpop_axis(1, (0, 0), axis=axis)
+
+        mesh = self.trimesh
+
+        try:
+            section = mesh.section(plane_origin=origin, plane_normal=normal)
+        except ImportError as import_err:
+            raise ImportError(
+                "Plotting 'CustomSurfaceMeshGeometry' requires 'networkx' and 'rtree'. "
+                "Please install the 'surfacemesh' dependencies. For example: "
+                "pip install 'tidy3d[surfacemesh]'."
+            ) from import_err
+
+        if section is None:
+            return []
+
+        # homogeneous transformation matrix to map to xy plane
+        mapping = np.eye(4)
+
+        # translate to origin
+        mapping[3, :3] = -np.array(origin)
+
+        # permute so normal is aligned with z axis
+        # and (y, z), (x, z), resp. (x, y) are aligned with (x, y)
+        identity = np.eye(3)
+        permutation = self.unpop_axis(identity[2], identity[0:2], axis=axis)
+        mapping[:3, :3] = np.array(permutation).T
+
+        try:
+            section2d, _ = section.to_planar(to_2D=mapping)
+            return section2d.polygons_full
+        except ImportError as import_err:
```

Comment:
made a comment earlier about importing networkx and rtree at the beginning of the file. But if these are only needed for this one method, I think how you've got it here is probably fine.